### ISSUE TITLE
Question about the affordance label

### ISSUE THREAD

**Issue by zdchan** [GentlesJan/AffordPose#2]
Hi,

Thanks for your great dataset. But may I ask how can I find the affordance label? It seems every object looks the same inside MeshLab regardless of the affordance. And the object_mesh vertexes look the same without any annotation about affordance. E.g., jar 4118 looks like this

![image](https://github.com/GentlesJan/AffordPose/assets/26100670/cdc3a5f5-709b-4814-b6d3-7d4139758e45)

I noticed every vertex is followed by 5 or 7. Does this means the affordance label?

Best,
Hui

**Comment by GentlesJan** [comment]
Yup, the fourth column in ‘object_mesh’ indicates the affordance label for every point of the object. Here’s a breakdown of the relationships: 1: Handle-grasp, 3: Press, 4: Lift, 5: Wrap-grasp, 6: Twist, 7: Support, 8:  Pull, and 9: Lever.

**Comment by GentlesJan** [comment]
> Hi,
> 
> Thanks for your great dataset. But may I ask how can I find the affordance label? It seems every object looks the same inside MeshLab regardless of the affordance. And the object_mesh vertexes look the same without any annotation about affordance. E.g., jar 4118 looks like this
> 
> ![image](https://user-images.githubusercontent.com/26100670/279755286-cdc3a5f5-709b-4814-b6d3-7d4139758e45.png)
> 
> I noticed every vertex is followed by 5 or 7. Does this means the affordance label?
> 
> Best, Hui

Yup, the fourth column in ‘object_mesh’ indicates the affordance label for every point of the object. Here’s a breakdown of the relationships: 1: Handle-grasp, 3: Press, 4: Lift, 5: Wrap-grasp, 6: Twist, 7: Support, 8: Pull, and 9: Lever.

**Comment by zdchan** [comment]
Hi,

Thanks for the answer. So the object_mesh in different json files for different hand poses and affordance categories of the same object are the same? E.g., all the json filea under bottle/3415 have the same object_mesh? And each point of the object mesh belongs to a single affordance label?

Best,
Hui

**Comment by GentlesJan** [comment]
> Hi,
> 
> Thanks for the answer. So the object_mesh in different json files for different hand poses and affordance categories of the same object are the same? E.g., all the json filea under bottle/3415 have the same object_mesh? And each point of the object mesh belongs to a single affordance label?
> 
> Best, Hui

Yes, that's right.

**Comment by zdchan** [comment]
Do you know how to visualize the mesh with different colors assigned to different affordance labels, just as your rendered images here? 

Another question is that, what does affordance label 0 mean?

 Thanks a lot!

![image](https://github.com/GentlesJan/AffordPose/assets/26100670/227d0119-2c5a-40d7-9664-e5b6900d3c6b)

**Comment by Cocolyl** [comment]
> Do you know how to visualize the mesh with different colors assigned to different affordance labels, just as your rendered images here?
> 
> Another question is that, what does affordance label 0 mean?
> 
> Thanks a lot!
> 
> ![image](https://private-user-images.githubusercontent.com/26100670/279950626-227d0119-2c5a-40d7-9664-e5b6900d3c6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4NzA2ODAsIm5iZiI6MTcwNDg3MDM4MCwicGF0aCI6Ii8yNjEwMDY3MC8yNzk5NTA2MjYtMjI3ZDAxMTktMmM1YS00MGQ3LTk2NjQtZTViNjkwMGQzYzZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDA3MDYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliMTI0ZjlmZjc3NjgyZTMwMDk1Mjk3ZTAwYzMyOWRiZjYwNTVmMDA0YmExYzA2ZThjYzNhMTZlYmIzNTk2MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-UN0Tn355-HuPjfR2IEq3M57Jyed7EjGmgKTCfohOJo)

I guess 0 means this point doesn't join the interaction.
